releases/release-5.1.2.md

```diff
@@ -0,0 +1,151 @@
+    - Fix the issue that a wrong selection is pushed down when the HAVING condition is used in the aggregate function [#26496](https://github.com/pingcap/tidb/issues/26496)
+    - Fix a bug that the query may return wrong results when the hash partition table deals with unsigned data [#26569](https://github.com/pingcap/tidb/issues/26569)
+    - Fix the unexpected behavior when casting an invalid string to DATE [#26762](https://github.com/pingcap/tidb/issues/26762)
+    - Fix a bug that creating partition fails if `NO_UNSIGNED_SUBTRACTION` is set [#26765](https://github.com/pingcap/tidb/issues/26765)
+    - Fix the issue that the distinct flag is missing when Apply is converted to Join [#26958](https://github.com/pingcap/tidb/issues/26958)
+    - Fix the issue that NO_ZERO_IN_DATE does not work on the default values [#26766](https://github.com/pingcap/tidb/issues/26766)
+    -  Set a block duration for the newly recovered TiFlash node to avoid blocking queries during this time [#26897](https://github.com/pingcap/tidb/pull/26897)
+    - Fix a bug that might occur when the CTE is referenced more than once [#26212](https://github.com/pingcap/tidb/issues/26212)
+    - Fix a CTE bug when MergeJoin is used [#25474](https://github.com/pingcap/tidb/issues/25474)
+    - Fix a bug that the 'SELECT FOR UPDATE' statement does not correctly lock the data when a normal table joins a partition table [#26251](https://github.com/pingcap/tidb/issues/26251)
+    - Fix the issue that the 'SELECT FOR UPDATE' statement returns an error when a normal table joins a partition table [#26250](https://github.com/pingcap/tidb/issues/26250)
+    - Fix the issue that `PointGet` does not use the lite version of resolve lock [#26562](https://github.com/pingcap/tidb/pull/26562)
+
++ TiKV
+
+    - Fix bug when TiKV upgrade from 3.0 to 4.x and 5.x but there are still some files left by import. [#10902](https://github.com/tikv/tikv/issues/10902)
+    - RaftStore Snapshot GC fix: fix the issue that snapshot GC missed GC snapshot files when there's one snapshot file failed to be GC-ed. [#10813](https://github.com/tikv/tikv/issues/10813)
+    - TiKV coprocessor slow log will only consider time spent on processing the request. Drop log instead of blocking threads when slogger thread is overloaded and queue is filled up. [#10865](https://github.com/tikv/tikv/pull/10865)
+    - Bug fix: fix an unexpected panic when exceeds deadline on processing copr requests. [#10852](https://github.com/tikv/tikv/issues/10852)
+    - Fix TiKV panic when enable Titan and upgrade from pre-5.0 version. Fix newer TiKV can't rollback to 5.0.x [#10842](https://github.com/tikv/tikv/pull/10842)
+    - Fix TiKV delete files before it ingests to RocksDB. [#10438](https://github.com/tikv/tikv/issues/10438)
+    - Fix the resolve failures caused by the left pessimisic locks. [#26404](https://github.com/pingcap/tidb/issues/26404)
```

Comment:
```suggestion
+ TiKV

    - Fix a bug that some files are missed to be imported during the process of importing snapshot files when upgrading TiKV from v3.x to v4.x or v5.x [#10902](https://github.com/tikv/tikv/issues/10902)
    - Fix the issue that the GC failure (such as file corrupted) of a single snapshot file stops the GC process of all other GC-able files [#10813](https://github.com/tikv/tikv/issues/10813)
    - The slow log of TiKV coprocessor only considers the time spent on processing requests [#10841](https://github.com/tikv/tikv/issues/10841)
    - Drop log instead of blocking threads when the slogger thread is overloaded and the queue is filled up [#10841](https://github.com/tikv/tikv/issues/10841)
    - Fix a bug of unexpected Panic caused by timeout when processing Coprocessor requests [#10857](https://github.com/tikv/tikv/pull/10857)
    - Fix the TiKV panic issue that occurs when upgrading from a pre-5.0 version with Titan enabled [#10842](https://github.com/tikv/tikv/pull/10842)
    - Fix the issue that TiKV of a newer version cannot be rolled back to v5.0.x [#10842](https://github.com/tikv/tikv/pull/10842)
    - Fix the issue that TiKV deletes files before the files are ingested to RocksDB [#10438](https://github.com/tikv/tikv/issues/10438)
    - Fix the parsing failure caused by the left pessimistic locks [#26404](https://github.com/pingcap/tidb/issues/26404)
```